pkg/metrics/pushclient/pushclient.go

```diff
@@ -0,0 +1,42 @@
+package pushclient
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// PushClient is used to send Prometheus metrics objects to any Prometheus
+// Push Gateway. It stores the URL and the client that will be used to push
+// to the desired gateway.
+type PushClient struct {
+	URL    string
+	Client *http.Client
+}
+
+// pushAllToAggregationGateway is a helper that takes care of the actual code that pushes to the prometheus
+// aggregation gateway. It takes a list of collectors and pushes all of them to the desired url.
+func (p *PushClient) pushAllToAggregationGateway(collectors ...prometheus.Collector) error {
+	allJobsName := "installer_metrics"
```

Comment:
1. I think we need to confirm what this value should be.
2. I think it makes sense to also move this value to a field in PushClient, which would allow us to defer 1 for now.